app/controllers/devise/api/tokens_controller.rb

```diff
@@ -10,6 +10,13 @@ class TokensController < Devise.api.config.base_controller.constantize
 
       # rubocop:disable Metrics/AbcSize
       def sign_up
+        unless Devise.api.config.sign_up.enabled
+          error_response = Devise::Api::Responses::ErrorResponse.new(request, error: :sign_up_disabled,
+                                                                     resource_class: resource_class)
+
+          return render json: error_response.body, status: error_response.status
```

Comment:
I think we don't need to custom error mesage if registration disabled

Reply:
I think it's not mandatory but it's more consistent with what was already done with the refresh_token_disabled error.